plugin/src/gatsby-node.ts

```diff
@@ -233,25 +233,23 @@ async function sourceChangedNodes(
 
   const { fetchDestroyEventsSince } = eventsApi(pluginOptions);
   const destroyEvents = await fetchDestroyEventsSince(lastBuildTime);
+
+  gatsbyApi.reporter.info(
+    `${destroyEvents.length} items have been deleted since ${lastBuildTime}`
+  );
+
   if (destroyEvents.length) {
-    for (const nodeType of shopifyNodeTypes) {
-      gatsbyApi.getNodesByType(nodeType).forEach((node) => {
-        /* This is currently untested because all the destroy events for the
-         * swag store are for products that this POC has never sourced!
-         *
-         * Also to consider: what about cascade delete? If a product is removed
-         * here, do we clean up variants, metafields, images, etc?
-         */
-        const event = destroyEvents.find(
-          (e: { subject_id: number; subject_type: string }) =>
-            e.subject_id === parseInt(node.shopifyId as string, 10) &&
-            node.internal.type === `Shopify${e.subject_type}`
-        );
-        if (event) {
-          gatsbyApi.actions.deleteNode(node);
-        }
-      });
-    }
+    gatsbyApi.reporter.info(`Removing matching nodes from Gatsby`);
+    destroyEvents.forEach((e) => {
+      const node = gatsbyApi
+        .getNodesByType(`Shopify${e.subject_type}`)
+        .find((n) => e.subject_id === parseInt(n.shopifyId as string, 10));
```

Comment:
This is still pretty expensive. It looks like [getNode](https://www.gatsbyjs.com/docs/reference/config-files/node-api-helpers/#getNode) would work here but I might be missing something.

Reply:
I think the problem with `getNode` is that it takes an ID. I don't really have the node ID here but rather the `shopifyId`. I agree it's probably still quite expensive and would be happy for a better strategy.

Reply:
You should be able to use the same createNodeId helper here as it's deterministic and always results in the same output given the same input.

Reply:
Excellent advice thank you

Reply:
Anytime!

Reply:
Updated!

Reply:
Looks great 😄